### ISSUE TITLE
How to debug custom loss function

### ISSUE THREAD

**Issue by jamesseeman** [keras-team/keras#13206]
This is likely a simple question as I'm a novice with Keras. I'm writing a custom loss function and I would like to debug it. How can I do this? The print() function doesn't work and loss/metric functions don't take any arguments beyond y_true and y_pred.

**Comment by bmxitalia** [comment]
Hi @jamesseeman, I have the same problem with Keras at the moment. The problem is that the loss function is given to the model with the add_loss method or with the parameter loss= of the compile method. When the model is compiled a compiled version of the loss is used during training. It seems that even if you have a K.print_tensor() instruction inside the loss, when the loss is compiled the model doesn't need to eval that during training. I know, it's frustrating. This is what I found.

**Comment by deeptavker** [comment]
@bmxitalia Hi, I am having a similar problem. I am trying to implement Elastic weight consolidation in keras and in order to do that, I need a loss function which uses model trainable weights to calculate loss, so I need it such that every time weights are updated, different loss values should be encountered. However, I am not being able to get that. It might be because of what you said, that while training, a compiled version of loss is used. 

Apart from that, I also haven't been able to debug the custom loss function. 

Here's my custom loss function. 

```python

def custom_loss(fisher, star_vars, lamda):

    def loss(y_true, y_pred):
       # here "model" is a global variable
        var_list = []
        for layer in model.layers:
            for weights in layer.get_weights():
                var_list.append(weights)

        ewc_loss = keras.losses.categorical_crossentropy(y_true, y_pred, from_logits=False, label_smoothing=0)
        for v in range(len(fisher)):
            ewc_loss += (lamda / 2.0) * tf.reduce_sum(tf.multiply(fisher[v].astype(np.float32),
                                                                   tf.square(var_list[v] - star_vars[v])))
            
        return ewc_loss
    return loss
```
Any ideas on how to work around this?

**Comment by painreign** [comment]
You can call the function yourself and see what it returns.
